### DISCUSSION TITLE
Too many devices earnapp linked on someone

### DISCUSSION THREAD

**Discussion by hungf1511** [engageub/InternetIncome#63] [Q&A]
The urls generated are linked to someone. When using 50 or more proxies

**Comment by engageub** [comment]
Hi,
Yes, this is a known issue which has come up recently due to the increase in number of devices used by users.
You will need to store your nodes from earnapp.txt file. Take a copy of the lines which were successful. 
Then replace in your earnapp.txt file. Then start it again.
This way you only use your device ids attached to your earnapp dashboard. 


There is another version for earnapp but it requires privileged permissions which is usually not recommended as it provides access to system resources. 
https://github.com/fazalfarhan01/EarnApp-Docker

Thank you

**Comment by hungf1511** [comment]
I still don't quite understand your instructions

**Reply by engageub** [reply]
I am rephrasing them below.
The script stores the nodes in earnapp.txt file in the same folder.
The node ids are generated before and added to earnapp without checking if it already exists or other user is using it.
In order to resolve your problem, you should delete the node ids from earnapp.txt which are not working and delete and start InternetIncome script. You have to do this until all your proxies are fulfilled with your earnapp nodes.

Steps to be followed.
1. Start InternetIncome script.
2. Add your node ids from earnapp.txt to earnapp dashboard
3. Delete the lines from earnapp.txt which are not working.
4. Stop the InternetIncome script using --delete command
5. Go back to step 1 and repeat the process.


This is time consuming based on the availability of nodes.
However the other version of earnapp will be deployed to test branch soon to avoid this manual generation of nodes.

**Comment by engageub** [comment]
Random ID generation logic has been changed in test branch for earnapp
You may try this branch and see if the issue still persists.

https://github.com/engageub/InternetIncome/blob/test/

Thank you

**Comment by engageub** [comment]
Random Id generation logic for earnapp has been updated in main branch.

https://github.com/engageub/InternetIncome


Thank you
